### ISSUE TITLE
Bluetooth connection error on iOS with ESP32 device ("Authentication is Insufficient" or "Encryption is insufficient")

### ISSUE THREAD

**Issue by Jucy-Dev** [espressif/esp-idf-provisioning-ios#80]
### Description:

I am experiencing an error when trying to connect to my ESP32 device using the iOS app **ESP BLE Prov**. The error message I receive is either "Authentication is Insufficient" or "Encryption is insufficient".

However, the connection works perfectly fine on the Android app.

### Frequency:

It happens every time. Sometimes is "Authentication is Insufficient » or otherwise "Encryption is insufficient » error.

### To reproduce:
Steps to reproduce the error:
- Launch the iOS app ESP BLE prov.
- Click on « I don’t have a QR code » to scan nearby bluetooth devices.
- Try to connect to the device via Bluetooth.
- The "Authentication is Insufficient" or "Encryption is insufficient" error is displayed.

### iOS information:
- Device type/model: iPhone 11
- iOS version: iOS 17.3.1
- iOS app version: v2.1.0

### Android information:
- Device model: realme 9i
- Android version: Android 13
- Android app version: v2.1.4

### Additional provisioning information:
- ESP Device type: ESP32 device
- Device type: BLE
- Security: Unsecure

### Screenshots:

Android connexion success: [Video example](https://github.com/espressif/esp-idf-provisioning-ios/assets/44801762/d557a190-eddc-4a90-9fbd-5235425f9cfb)
IOS connexion error: [Video example](https://github.com/espressif/esp-idf-provisioning-ios/assets/44801762/6d3bd577-85e4-417b-8b49-24a27c2dcde3)

### Request:

I would like to know if this is a known issue on iOS app and detailed reasons why it does this. If possible, please provide instructions on how to resolve this issue.

**Comment by blackcodefr** [comment]
Hi,

Did anyone look at this yet ?

I seems like a fairly obvious bug since the Android library and Android sample app work.

**Comment by vikas-chandra-mnnit** [comment]
Yes @blackcodefr, we are looking into this issue and will provide a fix soon.

**Comment by vikas-chandra-mnnit** [comment]
@Jucy-Dev @blackcodefr Unfortunately we are not able to reproduce this issue at our end. I have tried on ESP-IDF version 5.4. Please let us know your IDF version and any additional example configurations.

**Comment by vikas-chandra-mnnit** [comment]
@Jucy-Dev @blackcodefr Unfortunately we were not able to reproduce this issue at our end. We used the example from ESP-IDF version 5.4. Can you please let us know the IDF version and any additional steps to reproduce it?

**Comment by patafx** [comment]
The version of ESP-IDF is 4.4 and 4.4.7

**Comment by ajay-gantayet** [comment]
HI @patafx  We tried to reproduce the issue in the version 4.4.7 but we are still not able to reproduce it. Could you please send us the device logs in debug mode, so we can identify the root cause.

**Comment by MattBerg1953** [comment]
HI the first step I did on my iPhone 13 was to see if the BT device name was shown in the list. And it was missing. Did the same with Windows 11 laptop and it showed up, I could make a connection and send messages. 
My device used was the ESP32 Devkit V1 OT101. So my conclusion is that a deeper dive is needed into the way iPhone connects level, speed and timeouts.

**Comment by serdartoren** [comment]
https://github.com/espressif/esp-idf/tree/master/examples/bluetooth/bluedroid/ble/gatt_security_server
I've been trying to resolve this issue for 3 days. But I didn't fix it.

I am sending the example link to the relevant friend so that he can see the problem with his own eyes. 
Try this with iPhone 14 and you will see similar bug.
By the way, the same situation also exists in nimble

**Comment by serdartoren** [comment]
As you know, it is currently working without a password. 
I spent my last days trying to match it with the password. 
But there is a strange problem on the iPhone side.

**Comment by evan1715** [comment]
Any updates on this issue? Still experiencing the issue.

**Comment by vikas-chandra-mnnit** [comment]
> Any updates on this issue? Still experiencing the issue.

Hi @evan1715, Could you please provide the following details to help us debug the issue on our end?

- iOS Version
- App Version
- IDF version
- Firmware Logs

**Comment by serbyxp** [comment]
I been able to get it to pair bond... connect do HID over BLE with keyboard and mouse composite (iOS you need to set the device appearance to 962 or to mouse-pointer if you want it to show up as a touch device om the accessibility touch devices)... there are alot of quirks, the issue Im having is figuring out how to get it to re-pair when it disconnects or when you click the device in iOS bluetooth menu ( its in the paired list as not connected) when i click on it to connect it just fails, but it does pair and work just need to always forget the device and repair which is what im trying to resolve.

iOS seems very picky on how you advertise, mine had to be secure or encrypted to get it to pair. it would connect, but wouldn't show the pair dialog you get in your window without it I think even the MTU settings when pairing matters at least I've seen it mentioned... the chatgpt5 cant figure it out either btw

**Comment by vchan023** [comment]
@serbyxp wanted to clarify, you were able to connect over BLE with security type 0 on iOS? Can you provide more details?

On my end its very inconsistent, almost as if it works once then never again on a device. Always works on Android however.
